### DISCUSSION TITLE
How to get a up-to-state watched value in onChange callback of register?

### DISCUSSION THREAD

**Discussion by hyungjunk** [react-hook-form/react-hook-form#8000] [Q&A]
At version 6, `watch` triggered rerender, but after migrating to v7, `watch` is not triggering rerender on input change anymore and a method that should be called on every input change is not being called.

I have added `onChange` option to the register like below, the method is called but the watched value is not up-to-state but it keeps getting the last rendered value.

```
const checkDiff = () => {
    const currentForm = watch('myFieldArray');
    console.log({ currentForm })
    ...
}

<MyCustomInput
    {...register(`myFieldArray.${index}.name`, {
      onChange: checkDiff,
    })}
```

When I logged the watched value in the `onChange` input, the current watched value is logged, which is the result I want.
But the behavior was not what I expected so I put a breakpoint on `console.log({ currentForm })` line, then I found the watched value is not the current state but a result of one render behind.

For example, if I type `A`, `S`, `D`  sequentially on the input, logged results are `'', 'A', 'AS'`. 

It is very weird that this logging behavior happens depending on whether the debugger is set or not. 

Is there a way to get the latest value of input in this case?

**Comment by bluebill1049** [comment]
can you share a codesandbox?
